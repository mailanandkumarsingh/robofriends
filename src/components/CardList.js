import React from 'react';
import Card from './Card';

export default function CardList({robots}) {
    return (
        <div className="mw5 mw7-ns center">
            {
                robots.map((robot) => {
                    return <Card key ={robot.id} name={robot.name} email={robot.email} title={robot.username} id={robot.id}/>
                })
            }
        </div>
    )
}
