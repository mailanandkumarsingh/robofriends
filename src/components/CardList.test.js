import React from 'react';
import {shallow} from 'enzyme';
import CardList from './CardList';

it('tests for CardList shallow rendering', () => {
    const mockRobot = [{
        id: 1,
        title: 'RoboCop',
        email: 'rbc@gmail.com',
        username: 'roborob'
    }]
    expect(shallow(<CardList robots={mockRobot}/>)).toMatchSnapshot();
});