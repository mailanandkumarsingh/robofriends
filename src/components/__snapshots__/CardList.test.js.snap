// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for CardList shallow rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "rbc@gmail.com",
              "id": 1,
              "title": "RoboCop",
              "username": "roborob",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="rbc@gmail.com"
          id={1}
          name={undefined}
          title="roborob"
/>,
      ],
      "className": "mw5 mw7-ns center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "rbc@gmail.com",
          "id": 1,
          "name": undefined,
          "title": "roborob",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="rbc@gmail.com"
            id={1}
            name={undefined}
            title="roborob"
/>,
        ],
        "className": "mw5 mw7-ns center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "rbc@gmail.com",
            "id": 1,
            "name": undefined,
            "title": "roborob",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
